src/Hourai/Standard/Standard.cs

```diff
@@ -1,300 +1,355 @@
+      using (var sha = SHA384.Create()) {
+        await HashMessage(input, sha);
+      }
+    }
+
+    [Command("sha512")]
+    [Alias("sha-512")]
+    public async Task Sha512([Remainder] string input = "") {
+      using (var sha = SHA512.Create()) {
+        await HashMessage(input, sha);
+      }
+    }
+
+    static string GetURL(HouraiContext context, string input) {
+      return context.Message.Embeds.SingleOrDefault()?.Url ??
+        context.Message.Attachments.SingleOrDefault()?.Url;
+    }
+
+    static async Task<string> GetHash(string input, string url, HashAlgorithm hashing) {
+      byte[] hash;
+      if (url == null) {
+        hash = hashing.ComputeHash(Encoding.ASCII.GetBytes(input));
+      } else {
+        using (var httpClient = new HttpClient()) {
+          using (var request = new HttpRequestMessage(HttpMethod.Get, new Uri(url))) {
+            using (Stream contentStream = await (await httpClient.SendAsync(request)).Content.ReadAsStreamAsync()) {
+              hash = hashing.ComputeHash(contentStream);
+            }
+          }
+        }
+      }
+      return Convert.ToBase64String(hash);
+    }
+
+    async Task HashMessage(string input, HashAlgorithm hashing) {
+      var url = GetURL(Context, input);
+      await RespondAsync(await GetHash(input, url, hashing));
+    }
+
+    [Group("check")]
+    public class Verify : HouraiModule {
+
+      [Command("md5")]
+      public async Task Md5(string hash, [Remainder] string input = "") {
+        using (var md5 = MD5.Create()) {
+          await CheckMessage(input, hash, md5);
+        }
+      }
+
+      [Command("sha1")]
+      [Alias("sha-1")]
+      public async Task Sha1(string hash, [Remainder] string input = "") {
+        using (var sha = SHA1.Create()) {
+          await CheckMessage(input, hash, sha);
+        }
+      }
+
+      [Command("sha256")]
+      [Alias("sha-256")]
+      public async Task Sha256(string hash, [Remainder] string input = "") {
+        using (var sha = SHA256.Create()) {
+          await CheckMessage(input, hash, sha);
+        }
+      }
+
+      [Command("sha384")]
+      [Alias("sha-384")]
+      public async Task Sha384(string hash, [Remainder] string input = "") {
+        using (var sha = SHA384.Create()) {
+          await CheckMessage(input, hash, sha);
+        }
+      }
+
+      [Command("sha512")]
+      [Alias("sha-512")]
+      public async Task Sha512(string hash, [Remainder] string input = "") {
+        using (var sha = SHA512.Create()) {
+          await CheckMessage(input, hash, sha);
+        }
+      }
+
+      async Task CheckMessage(string input, string hash, HashAlgorithm hashing) {
+        var url = GetURL(Context, input);
+        var hashCheck = await GetHash(input, url, hashing);
+        if (hash == hashCheck)
+          await Success();
+        else {
+          await RespondAsync($":x: Hash `{hashCheck}` does not match `{hash}`");
+        }
+      }
+    }
+
+  }
+
+}
+
+}

```

Comment:
Why are all these files changed completely? The changes have been to only a few lines.